### PR TITLE
A few renamings and bug fixes

### PR DESCRIPTION
**Pull request by semi-h**
- memblock_t is renamed as field_t because we use allocator and memblock exclusively for allocating scalar fields.
- cudaallocator and cudafield(cudamemblock) is renamed with underscore, I think it looks better and easier to read
- The rest is just few small bug fixes

### CODE REVIEW

**Comment by tlestang** [comment]
@semi-h could you add a description stating what you changed and why?

File: src/vector3d.f90 [hunk 1]
```diff
@@ -12,7 +12,8 @@ module m_slab
       class(field_t), pointer :: u1, u2, u3
       real :: xnu
    contains
-      procedure(field_op), deferred :: transport, div
+      procedure(field_op), deferred :: transport
+      procedure(field_op), deferred :: div
```
Comment (tlestang) [inline src/vector3d.f90:16]: That's to make nvfortran happy if I remember correctly?

File: tests/test_diffengine.f90 [hunk 2]
```diff
@@ -7,22 +7,29 @@ program test_diffengine
...
+  allocate(f_vec(SZ,n))
+  do j = 1, n
+     do i = 1, SZ
+        f_vec(i, j) = sin((j-1)*dx)
+     end do
+  end do
 
   ! First derivative with periodic boundary conditions
-  expected = [(cos((i-1)*dx), i=1,n)]
-  expected_vec = transpose(reshape(expected, [n, SZ], pad = expected))
```
Comment (tlestang) [inline tests/test_diffengine.f90:25]: Was it not correct? In any case replacing this by a double loop is probably for the best.

**Review by tlestang — APPROVED** [review]

File: src/vector3d.f90 (hunk 1 above)
Comment (semi-h) [reply src/vector3d.f90:16]: Yes, it doesn't compile when you list them in the same line

File: tests/test_diffengine.f90 (hunk 2 above)
Comment (semi-h) [reply tests/test_diffengine.f90:25]: My nvfortran compiler was giving an error here for some reason
